File: Serene/Serene.Script/Common/Navigation/ThemeSelection.cs
namespace Serene.Common
{
    using jQueryApi;
    using Serenity;
    using System.Linq;

    public class ThemeSelection : Widget
    {
        public ThemeSelection(jQueryObject select)
            : base(select)
        {
            var self = this;

            this.Change(e =>
            {
                jQuery.Instance.cookie("ThemePreference", select.GetValue(), new {
                    path = Q.Config.ApplicationPath,
                    expires = 365
                });
                J("body").RemoveClass("skin-" + GetCurrentTheme());
                J("body").AddClass("skin-" + select.GetValue());
            });

            Q.AddOption(select, "blue", Q.Text("Site.Layout.ThemeBlue"));
            Q.AddOption(select, "blue-light", Q.Text("Site.Layout.ThemeBlueLight"));
            Q.AddOption(select, "purple", Q.Text("Site.Layout.ThemePurple"));
            Q.AddOption(select, "purple-light", Q.Text("Site.Layout.ThemePurpleLight"));
            Q.AddOption(select, "red", Q.Text("Site.Layout.ThemeRed"));
            Q.AddOption(select, "red-light", Q.Text("Site.Layout.ThemeRedLight"));
            Q.AddOption(select, "green", Q.Text("Site.Layout.ThemeGreen"));
            Q.AddOption(select, "green-light", Q.Text("Site.Layout.ThemeGreenLight"));
            Q.AddOption(select, "yellow", Q.Text("Site.Layout.ThemeYellow"));
            Q.AddOption(select, "yellow-light", Q.Text("Site.Layout.ThemeYellowLight"));
            Q.AddOption(select, "black", Q.Text("Site.Layout.ThemeBlack"));
            Q.AddOption(select, "black-light", Q.Text("Site.Layout.ThemeBlackLight"));

            select.Value(GetCurrentTheme());
        }

        private string GetCurrentTheme()
        {
            var skinClass = (J("body").GetAttribute("class") ?? "").Split(' ')
                .FirstOrDefault(x => x.StartsWith("skin-"));

            if (skinClass != null)
                return skinClass.Substr(5);

            return "blue";
        }
    }
}

File: Serene/Serene.AspNetCore/Modules/Common/Navigation/ThemeSelection.ts
namespace Serene.Common {
    export class ThemeSelection extends Serenity.Widget<any> {
        constructor(select: JQuery) {
            super(select);

            this.change(e => {
                $.cookie('ThemePreference', select.val(), {
                    path: Q.Config.applicationPath,
                    expires: 365
                });
                $('body').removeClass('skin-' + this.getCurrentTheme());
                $('body').addClass('skin-' + select.val());
            });

            Q.addOption(select, 'blue', Q.text('Site.Layout.ThemeBlue'));
            Q.addOption(select, 'blue-light', Q.text('Site.Layout.ThemeBlueLight'));
            Q.addOption(select, 'purple', Q.text('Site.Layout.ThemePurple'));
            Q.addOption(select, 'purple-light', Q.text('Site.Layout.ThemePurpleLight'));
            Q.addOption(select, 'red', Q.text('Site.Layout.ThemeRed'));
            Q.addOption(select, 'red-light', Q.text('Site.Layout.ThemeRedLight'));
            Q.addOption(select, 'green', Q.text('Site.Layout.ThemeGreen'));
            Q.addOption(select, 'green-light', Q.text('Site.Layout.ThemeGreenLight'));
            Q.addOption(select, 'yellow', Q.text('Site.Layout.ThemeYellow'));
            Q.addOption(select, 'yellow-light', Q.text('Site.Layout.ThemeYellowLight'));
            Q.addOption(select, 'black', Q.text('Site.Layout.ThemeBlack'));
            Q.addOption(select, 'black-light', Q.text('Site.Layout.ThemeBlackLight'));

            select.val(this.getCurrentTheme());
        }

        protected getCurrentTheme() {
            var skinClass = Q.first(($('body').attr('class') || '').split(' '), x => Q.startsWith(x, 'skin-'));
            if (skinClass) {
                return skinClass.substr(5);
            }

            return 'blue';
        }
    }
}